CRM/Donrec/Logic/Selector.php

```diff
@@ -23,12 +23,22 @@ class CRM_Donrec_Logic_Selector {
    *   creation result/error
    */
   public static function createSnapshot($values) {
+    // Process date picker values.
+    [
+      $values['donrec_contribution_horizon_from'],
+      $values['donrec_contribution_horizon_to']
+    ] = CRM_Utils_Date::getFromTo(
+      $values['donrec_contribution_horizon_relative'],
+      $values['donrec_contribution_horizon_from'],
+      $values['donrec_contribution_horizon_to']
+    );
+
```

Comment:
This processes relative dates and formats `from` and `to` values with `YmdHiS`, unfortunately, that's why we have to re-format again below.